tracing-subscriber/src/fmt/writer.rs

```diff
@@ -37,35 +108,91 @@ pub trait MakeWriter {
     /// [`FmtSubscriber`]: ../struct.Subscriber.html
     /// [`io::Write`]: https://doc.rust-lang.org/std/io/trait.Write.html
     /// [`MakeWriter`]: trait.MakeWriter.html
-    fn make_writer(&self) -> Self::Writer;
+    fn make_writer(&'a self) -> Self::Writer;
 }
 
-impl<F, W> MakeWriter for F
+/// A type implementing [`io::Write`] for a [`MutexGuard`] where tyhe type
+/// inside the [`Mutex`] implements [`io::Write`].
+///
+/// This is used by the [`MakeWriter`] implementation for [`Mutex`], because
+/// [`MutexGuard`] itself will not implement [`io::Write`] — instead, it
+/// _dereferences_ to a type implementing [`io::Write`]. Because [`MakeWriter`]
+/// requires the `Writer` type to implement [`io::Write`], it's necessary to add
+/// a newtype that forwards the trait implementation.
+///
+/// [`io::Write`]: https://doc.rust-lang.org/std/io/trait.Write.html
+/// [`MutexGuard`]: https://doc.rust-lang.org/std/sync/struct.MutexGuard.html
+/// [`Mutex`]: https://doc.rust-lang.org/std/sync/struct.Mutex.html
+/// [`MakeWriter`]: trait.MakeWriter.html
+#[derive(Debug)]
+pub struct MutexGuardWriter<'a, W>(MutexGuard<'a, W>);
+
+impl<'a, F, W> MakeWriter<'a> for F
 where
     F: Fn() -> W,
     W: io::Write,
 {
     type Writer = W;
 
-    fn make_writer(&self) -> Self::Writer {
+    fn make_writer(&'a self) -> Self::Writer {
         (self)()
     }
 }
 
+impl<'a, W> MakeWriter<'a> for Mutex<W>
+where
+    W: io::Write + 'a,
+{
+    type Writer = MutexGuardWriter<'a, W>;
+
+    fn make_writer(&'a self) -> Self::Writer {
+        MutexGuardWriter(self.lock().expect("lock poisoned"))
+    }
+}
+
+impl<'a, W> io::Write for MutexGuardWriter<'a, W>
+where
+    W: io::Write,
```

Comment:
Thinking aloud: should there be an equivalent implementation for `parking_lot`'s Mutex (under the same feature flag), given that the `tracing-subscriber` defaults to pulling in `parking_lot`?

Reply:
probably, yeah

Reply:
Can you add these feature flags now?

Reply:
i already did: 254a2f03d3986377fcc5ffbecf2794437049b7ce